From 8c0867fce0e2642d8f8ef005e88fffca685a6768 Mon Sep 17 00:00:00 2001
From: nicm <nicm>
Date: Tue, 7 Apr 2015 13:06:22 +0000
Subject: When replacing, don't free the old paste until after the new
 one's name has been copied. Fixes a use-after-free in window-copy.c.
 Bug reported by J Raynor (who also provided a different fix).

---
 paste.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/paste.c
+++ b/paste.c
@@ -246,9 +246,6 @@
 		return (-1);
 	}
 
-	pb = paste_get_name(name);
-	if (pb != NULL)
-		paste_free_name(name);
 
 	pb = xmalloc(sizeof *pb);
 
@@ -260,6 +257,9 @@
 	pb->automatic = 0;
 	pb->order = paste_next_order++;
 
+	if (paste_get_name(name) != NULL)
+		paste_free_name(name);
+
 	RB_INSERT(paste_name_tree, &paste_by_name, pb);
 	RB_INSERT(paste_time_tree, &paste_by_time, pb);
 
